src/icalendar/tests/test_unit_caselessdict.py

```diff
@@ -125,3 +125,40 @@ def test_CaselessDict(self):
 
         keys = sorted(ncd.keys())
         self.assertEqual(keys, ["KEY1", "KEY2", "KEY3", "KEY5", "KEY6"])
+
+    def test_eq_with_non_dict_types(self):
+        """Test that CaselessDict.__eq__ handles non-dict comparisons correctly."""
+        CaselessDict = icalendar.caselessdict.CaselessDict
+
+        d = CaselessDict()
+        d['test'] = 1
+
+        # Should return False for non-dict types, not crash
+        self.assertFalse(d == "string")
```

Comment:
We use pytest. Why do you use the unittest API? Pytest will generate nice messages for an assert statement.